lucene/core/src/java/org/apache/lucene/store/ByteBufferIndexInput.java

```diff
@@ -107,6 +118,28 @@ public final void readBytes(byte[] b, int offset, int len) throws IOException {
     }
   }
 
+  @Override
+  public void readLELongs(long[] dst, int offset, int length) throws IOException {
+    if (curLongBufferViews == null) {
+      // Lazy init to not make pay for memory and initialization cost if you don't need to read arrays of longs
```

Comment:
Maybe also explain that only certain index files (just postings today, but maybe doc values soon too) use the `read/writeLongs` APIs, justifying the laziness?

Reply:
Actually I was thinking of using it to read doc IDs in BKDReader next. I agree doc values should probably look into it as well.

Reply:
> under the hood the CPU must still do unaligned long decodes (which I think modern X86-64 are good at)? Maybe add a comment about why this trick is worthwhile?

This is mostly a workaround for the lack of ByteBuffer#getLongs(long[]). Avoiding the LongBuffer view would require to call ByteBuffer#getLong in a loop, which seems to have some per-long overhead according to the microbenchmarks I ran. I'll leave a comment.